### PR TITLE
Stm32 cmake refactor

### LINKED ISSUE #12: [BUG] Configuration time takes ~6 mins with cmake

### PR DESCRIPTION
**Pull request by safesintesi**
Fixes #12 .

FetchContent with low connection would waste time downloading the libraries.
A Flag not for FAT would force it to download freeRTOS-kernel a second time leading to a waste of time.
Since those repos are already includes through submodules, downloading is redundant.
This fix takes the configuration time down from minutes to ~0.7s.

With the occasion I refactored the Cmake file structure to fix some link dependencies and some flags:
- Added targets for drivers and core
- Fixed some libraries link
- Defined STM32F411xE as default build in Drivers/CMSIS/.../stm32f4xx.h

This now builds with cmake but needs a tweek on FAT repo.
A pull request has been already made & changes are accepted and waiting to be merged:
https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/pull/59

Edit: Also needs Cmake version bump to 3.18

Edit:
- FAT pull request has been merged and now this repo has updated the submodule.
- We want to fix the Thread lib too since we're at it, so this won't be merged until we solve the issue.

### CODE REVIEW

**Comment by BIRSAx2** [comment]
As of now, your pull request to the FAT repo has not been merged. I'll wait to review it until then because, as it stands, the project doesn't build.

File: Core/Src/CMakeLists.txt
Comment (BIRSAx2) [inline Core/Src/CMakeLists.txt:1]: You did not include the Startup/startup_stm32f411xe.s file in the build, which specifies, among other things, the initial PC and sets the CPU in Thread mode. 
As a result, the build throws the following warning: ```warning: cannot find entry symbol Reset_Handler; defaulting to 08000000.```

Comment (safesintesi) [reply Core/Src/CMakeLists.txt:1]: Good catch, that file wasn't included even on the original cmake structure on stm32 branch. I'll fix it this afternoon.

**Comment by BIRSAx2** [comment]
I've reviewed the code, and it looks good overall. 
The only issues I found are already mentioned: the inclusion of "startup_stm32f411xe.s," and the dependency on the FAT repo's pull request. 
Once these are addressed, we can proceed with the merge. If there are delays with FAT repo, considering a fork might be a viable option. Thanks!

**Comment by safesintesi** [comment]
@BIRSAx2 Should be good now, can you please check?

**Comment by safesintesi** [comment]
@gekoxyz seems like this is not the only step to take to fix the Thread issue.
I suggest keeping this pull request for the original purpose (fixing the FetchContent redundancy) and opening a new one for the linux port.
To do this we should revert the last changes, before merging and closing the pull request.

File: Core/Src/task_gianluca.c
Comment (BIRSAx2) [inline Core/Src/task_gianluca.c:1]: I noticed that the file task_gianluca.h has been deleted in this pull request. However, the build doesn't compile without it. Could you please provide more information about the deletion of this file and how we can ensure that the build remains functional?

Comment (BIRSAx2) [reply Core/Src/task_gianluca.c:1]: On top of that, the previous two tasks Task1 and Task2 are not running when building with CMake (the led is not blinking), while they do when building with STM32CubeIDE

**Pull request closed without merging**